File: apps/api/app/services/audit.py
from datetime import datetime, timezone
from uuid import UUID

from sqlalchemy.orm import Session

from ..models import AuditEvent


def record_security_event(
    session: Session,
    *,
    event_type: str,
    organization_id: UUID | None,
    actor_user_id: UUID | None,
    action: str,
    result: str,
    metadata: dict | None = None,
) -> None:
    session.add(
        AuditEvent(
            organization_id=organization_id,
            actor_user_id=actor_user_id,
            event_type=event_type,
            resource_type="authentication",
            action=action,
            result=result,
            event_metadata=metadata,
            created_at=datetime.now(timezone.utc),
        )
    )

File: apps/web/src/types/audit.ts
/** apps/api/app/schemas/management.py::AuditEventRead */
export type AuditEvent = {
  id: string
  organization_id: string | null
  actor_user_id: string | null
  event_type: string
  resource_type: string
  resource_id: string | null
  action: string
  result: string
  event_metadata: Record<string, unknown> | null
  created_at: string
}
